src/rtc_pl031.rs

```diff
@@ -0,0 +1,639 @@
+            RTCDR => (), // Ingore writes to the data register.
+            RTCMR => {
+                // Set the match register, though this is not currently used.
+                // TODO: Implement the match register functionality.
+                self.mr = val;
+            }
+            RTCLR => {
+                // Writing to the load register adjusts both the load register
+                // and the counter to ensure that a write to RTCLR followed by
+                // an immediate read of RTCDR will return the loaded value.
+                self.counter = Instant::now();
+                self.lr = val;
+            }
+            RTCCR => {
+                // Writing 1 to the control register resets the RTC value,
+                // which means both the counter and load register are reset.
+                if val == 1 {
+                    self.counter = Instant::now();
+                    self.lr = 0;
+                }
+            }
+            RTCIMSC => {
+                // Set or clear the interrupt mask.
+                self.imsc = val & 1;
+            }
+            RTCRIS => (), // Ignore writes to the interrupt status register.
+            RTCMIS => (), // Ignore writes to the masked interrupt status register.
+            RTCICR => {
+                // Writing 1 clears the interrupt.
+                self.ris &= !val;
+            }
+            _ => {
+                return Err(Error::BadWriteOffset(offset));
+            }
+        }
+        Ok(())
+    }
+
+    /// Handles a read request from the driver at `offset` offset from the
+    /// base register address.
+    ///
+    /// # Arguments
+    /// * `offset` - The offset from the base register specifying
+    ///              the register to be read.
+    /// * `data` - The litte-endian, 4 byte array storing the read value.
+    ///
+    /// # Example
+    ///
+    /// You can see an example of how to use this function in the
+    /// [`Example` section from `RTC`](struct.RTC.html#example).
+    pub fn read(&mut self, offset: u64, data: &mut [u8; 4]) {
+        let v;
+        let mut read_ok = true;
+
+        if offset < AMBA_ID_HIGH && offset >= AMBA_ID_LOW {
+            let index = ((offset - AMBA_ID_LOW) >> 2) as usize;
+            v = u32::from(PL031_ID[index]);
+        } else {
+            v = match offset {
+                RTCDR => self.get_rtc_value(),
+                RTCMR => {
+                    // Read the match register, though this is not currently used.
+                    // TODO: Implement the match register functionality.
+                    self.mr
+                }
+                RTCLR => self.lr,
+                RTCCR => 1, // RTC is always enabled.
+                RTCIMSC => self.imsc,
+                RTCRIS => self.ris,
+                RTCMIS => self.ris & self.imsc,
+                _ => {
+                    // If the offset is invalid, do nothing.
+                    read_ok = false;
+                    0
```

Comment:
Can we just `return` here instead of setting the `read_ok`? It should have the same behavior.

Reply:
Yes.  Done.